extensions/SharkPool/SoundCloud-API.js

```diff
@@ -0,0 +1,363 @@
+        if (!response.ok) return "Error: 404";
+        let json = await response.json();
+        for (var i = 0; i < path.length; i++) {
+          json = json[path[i]];
+        }
+        if (path[0] === "duration") json = this.milli2Time(json);
+        return json;
+      } catch { return "" }
+    }
+    milli2Time(milli) {
+      let seconds = Math.floor(milli / 1000);
+      let hours = Math.floor(seconds / 3600);
+      seconds %= 3600;
+      let minutes = Math.floor(seconds / 60);
+      seconds %= 60;
+      return `${hours}:${String(minutes).padStart(2, "0")}:${String(seconds).padStart(2, "0")}`;
+    }
+
+    async getTrackMp3(args) {
+      try {
+        let response;
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/tracks/${args.ID}/download?client_id=${clientID}&app_locale=en`)) return "";
+        // eslint-disable-next-line
+        response = await fetch(`${proxy2}https://api-v2.soundcloud.com/tracks/${args.ID}/download?client_id=${clientID}&app_locale=en`);
+        if (!response.ok) return "Error: 404";
+        let json = await response.json();
+        return json.redirectUri ?? "Error: Downloads are Disabled for this Track";
+      } catch { return "" }
+    }
+
+    async getTrackComment(args) {
+      try {
+        let response;
+        const type = args.TYPE === "new" ? "newest" : "oldest";
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/tracks/${args.ID}/comments?sort=${type}&threaded=1&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`)) return "";
+        // eslint-disable-next-line
+        response = await fetch(`${proxy}https://api-v2.soundcloud.com/tracks/${args.ID}/comments?sort=${type}&threaded=1&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`);
+        if (!response.ok) return "Error: 404";
+        let json = await response.json();
+        return JSON.stringify(json.collection);
+      } catch { return "[]" }
+    }
+
+    async searchTracks(args) {
+      try {
+        let response;
+        args.QUERY = encodeURIComponent(args.QUERY);
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/search/tracks?q=${args.QUERY}&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`)) return "";
+        // eslint-disable-next-line
+        response = await fetch(`${proxy}https://api-v2.soundcloud.com/search/tracks?q=${args.QUERY}&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`);
+        if (!response.ok) return "Error: 404";
+        let json = await response.json();
+        return JSON.stringify(json.collection);
+      } catch { return "[]" }
+    }
+
+    async getArtistAtt(args) {
+      try {
+        let response;
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/users/${args.ID}?client_id=${clientID}&app_locale=en`)) return "";
+        // eslint-disable-next-line
```

Comment:
Use `Scratch.fetch` or write a comment why you can't. It's important to know for maintenance.